.github/workflows/build-test.yaml

```diff
@@ -182,7 +184,7 @@ jobs:
       - uses: "actions/setup-go@v3"
         with:
           go-version: "${{ env.GO_VERSION }}"
-          cache: "true"
+          cache: "false" # do not cache to prevent cache poisoning
```

Comment:
Can you add the explicit cache false below too and document?

Reply:
done: https://github.com/authzed/spicedb/pull/961/commits/f0327e7c73813a5d53ee4b22843a1f164190b94e